From 61a8034c9906d1afbbe4a592f15589f27c3f058e Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Tue, 26 Mar 2024 10:30:44 +0530
Subject: [PATCH] Adding systemlock object path interface and systemlocked
 property

To Disabling set commands when systemlock property is enabled

jira link: https://oneami.atlassian.net/browse/OT-3317

Tested:

root@AMIOT-821275a9c272:~# busctl introspect xyz.openbmc_project.Settings /xyz/openbmc_project/control/systemlock
NAME                                            TYPE      SIGNATURE RESULT/VALUE FLAGS
org.freedesktop.DBus.Introspectable             interface -         -            -
.Introspect                                     method    -         s            -
org.freedesktop.DBus.Peer                       interface -         -            -
.GetMachineId                                   method    -         s            -
.Ping                                           method    -         -            -
org.freedesktop.DBus.Properties                 interface -         -            -
.Get                                            method    ss        v            -
.GetAll                                         method    s         a{sv}        -
.Set                                            method    ssv       -            -
.PropertiesChanged                              signal    sa{sv}as  -            -
xyz.openbmc_project.Control.Security.systemLock interface -         -            -
.SystemLocked                                   property  b         false        emits-change writable

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 include/defaults.hpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index f1b258e..674124c 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -91,6 +91,12 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
 
     setting->addProperty("PowerRestoreDelay", static_cast<uint64_t>(0));
 
+    setting = &settings.emplace_back(
+        objectServer, "/xyz/openbmc_project/control/systemlock",
+        "xyz.openbmc_project.Control.Security.SystemLock");
+
+    setting->addProperty("SystemLocked", false);
+
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/host0/acpi_power_state",
         "xyz.openbmc_project.Control.Power.ACPIPowerState");
-- 
2.25.1

